### PR TITLE
DP-516 BE- Share my information - Data sharing endpoint (API) - Form answers

### PR DESCRIPTION
**Pull request by glgoaco**
[DP-516](https://noticingsystem.atlassian.net/browse/DP-516)

The Data Sharing API endpoint returns supplier information for a valid share code with a 200 OK response or a 404 error if the share code does not exist.

### CODE REVIEW

File: Services/CO.CDP.OrganisationInformation.Persistence/Forms/FormAnswer.cs [hunk 1]
```diff
@@ -20,6 +20,7 @@ public class SharedConsent : IEntityDate
     public DateTimeOffset? SubmittedAt { get; set; }
     public required string FormVersionId { get; init; }
     public string? BookingReference { get; set; }
+    public string? ShareCode { get; set; }
```
Comment (dharmverma) [inline Services/CO.CDP.OrganisationInformation.Persistence/Forms/FormAnswer.cs:23]: Is ShareCode different than BookingReference?

Comment (JBaigGoaco) [reply Services/CO.CDP.OrganisationInformation.Persistence/Forms/FormAnswer.cs:23]: Good Point @dharmverma  
we have a share code in BookingReference
image

Comment (jakzal) [reply Services/CO.CDP.OrganisationInformation.Persistence/Forms/FormAnswer.cs:23]: It is not. We already have BookingReferecen, so either use BookingReference, or replace BookingReference with ShareCode.

If you decided to replace BookingReference with ShareCode, I'd prefer to have it done in a separate PR as it impacts few other things in progress.

File: Services/CO.CDP.Forms.WebApi.Tests/UseCase/UpdateFormSectionAnswersUseCaseTest.cs [hunk 2]
```diff
@@ -148,8 +148,6 @@ public async Task Execute_ShouldCreateNewSharedConsent_WhenNoDraftSharedConsentE
         var command = (formId: section.Form.Guid, sectionId: section.Guid, answerSetId,
             organisationId: organisation.Guid, answers);
 
-        GivenSharedConsentDoesNotExist(section.Form.Guid, organisation.Guid);
-
```
Comment (jakzal) [inline Services/CO.CDP.Forms.WebApi.Tests/UseCase/UpdateFormSectionAnswersUseCaseTest.cs:152]: Why's this removed?

Comment (glgoaco) [reply Services/CO.CDP.Forms.WebApi.Tests/UseCase/UpdateFormSectionAnswersUseCaseTest.cs:152]: When refactoring DatabaseFormRepository to move share methods to DatabaseShareCodeRepository, I noticed that UpdateFormSectionAnswersUseCase now uses GetSharedConsentDraftWithAnswersAsync, making the call to GetSharedConsentDraftAsync redundant.

**Review by dharmverma — APPROVED** [review]

**Review by maciej-goaco — APPROVED** [review]

**Pull request merged**